identities/tests.py

```diff
@@ -257,3 +268,124 @@ def test_create_identity_no_details(self):
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
         d = Identity.objects.last()
         self.assertEqual(d.version, 1)
+
+
+class TestOptOutAPI(AuthenticatedAPITestCase):
+    def test_create_opt_out_with_identity(self):
+        # Setup
+        identity = self.make_identity()
+        opt_out = TEST_OPTOUT
+        opt_out["identity"] = reverse('identity-detail',
+                                      kwargs={'pk': identity.pk})
+        # Execute
+        response = self.client.post('/api/v1/optout/',
+                                    json.dumps(opt_out),
+                                    content_type='application/json')
+        # Check
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        d = OptOut.objects.get(id=response.data["id"])
+        self.assertEqual(d.identity, identity)
+        self.assertEqual(d.request_source, "test_source")
+        self.assertEqual(d.requestor_source_id, '1')
+
+    def test_create_opt_out_with_address(self):
+        # Setup
+        identity = self.make_identity()
+        opt_out = TEST_OPTOUT
+        # Execute
+        response = self.client.post('/api/v1/optout/',
+                                    json.dumps(opt_out),
+                                    content_type='application/json')
+        # Check
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        d = OptOut.objects.get(id=response.data["id"])
+        self.assertEqual(d.identity, identity)
+        self.assertEqual(d.request_source, "test_source")
+        self.assertEqual(d.requestor_source_id, '1')
+
+    def test_create_opt_out_no_address(self):
+        # Setup
```

Comment:
This seems to be identical to `test_create_opt_out_with_address` (and `opt_out` currently includes an address). I'm guessing the intention was to not include the address and get an error response?